File: frontend/demo/component/app-layout/app-layout-bottom-navbar.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/app-layout';
import '@vaadin/app-layout/vaadin-drawer-toggle';
import '@vaadin/horizontal-layout';
import '@vaadin/icon';
import '@vaadin/icons';
import { css, html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';
import { patchAppLayoutNavigation } from './app-layout-helper'; // hidden-source-line

@customElement('app-layout-bottom-navbar')
export class Example extends LitElement {
  static override styles = css`
    h1 {
      margin: var(--vaadin-gap-m);
    }

    /* hidden-source-line: the bottom navbar is forced on in the example */
    vaadin-app-layout /* hidden-source-line */ {
      --vaadin-app-layout-touch-optimized: true; /* hidden-source-line */
    } /* hidden-source-line */
  `;

  /* prettier-ignore */ protected firstUpdated() { // hidden-source-line
    patchAppLayoutNavigation(this.shadowRoot!.querySelector('vaadin-horizontal-layout')!); // hidden-source-line
  } // hidden-source-line

  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <!-- --vaadin-app-layout-touch-optimized is only enforced as part of this example -->
      <vaadin-app-layout style="--vaadin-app-layout-touch-optimized: true">
        <h1 slot="navbar">MyApp</h1>
        <vaadin-horizontal-layout
          slot="navbar touch-optimized"
          style="width: 100%; justify-content: space-evenly;"
        >
          <a
            href="/dashboard"
            aria-label="Dashboard"
            style="display: flex; align-items: center; padding: 0 1.5rem;"
          >
            <vaadin-icon icon="vaadin:dashboard"></vaadin-icon>
          </a>
          <a
            href="/orders"
            aria-label="Orders"
            style="display: flex; align-items: center; padding: 0 1.5rem;"
          >
            <vaadin-icon icon="vaadin:cart"></vaadin-icon>
          </a>
          <!-- end::snippet[] -->
          <a
            href="/customers"
            aria-label="Customers"
            style="display: flex; align-items: center; padding: 0 1.5rem;"
          >
            <vaadin-icon icon="vaadin:user-heart"></vaadin-icon>
          </a>
          <a
            href="/products"
            aria-label="Products"
            style="display: flex; align-items: center; padding: 0 1.5rem;"
          >
            <vaadin-icon icon="vaadin:package"></vaadin-icon>
          </a>
          <!-- tag::snippet[] -->
        </vaadin-horizontal-layout>
        <div class="content">
          <h2>View title</h2>
          <p>View content</p>
        </div>
      </vaadin-app-layout>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/applayout/AppLayoutBottomNavbar.java
package com.vaadin.demo.component.applayout;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.applayout.AppLayout;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.H1;
import com.vaadin.flow.component.html.H2;
import com.vaadin.flow.component.html.Paragraph;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.router.RouterLink;
import com.vaadin.flow.theme.lumo.LumoUtility;

@Route("app-layout-bottom-navbar")
// tag::snippet[]
public class AppLayoutBottomNavbar extends AppLayout {

    public AppLayoutBottomNavbar() {
        getElement().getStyle().set("--vaadin-app-layout-touch-optimized", // hidden-source-line
                "true"); // hidden-source-line
        H1 title = new H1("MyApp");
        title.getStyle().set("font-size", "var(--lumo-font-size-l)")
                .set("margin", "var(--lumo-space-m) var(--lumo-space-l)");

        HorizontalLayout nav = getNavigation();
        nav.getElement().executeJs("window.patchAppLayoutNavigation(this);"); // hidden-source-line

        H2 viewTitle = new H2("View title");
        Paragraph viewContent = new Paragraph("View content");

        Div content = new Div();
        content.add(viewTitle, viewContent);

        addToNavbar(title);
        addToNavbar(true, nav);

        setContent(content);
    }
    // end::snippet[]

    private HorizontalLayout getNavigation() {
            HorizontalLayout navigation = new HorizontalLayout();
            navigation.addClassNames(LumoUtility.Width.FULL,
                            LumoUtility.JustifyContent.EVENLY,
                            LumoUtility.AlignSelf.STRETCH);
            navigation.setPadding(false);
            navigation.setSpacing(false);
            navigation.add(createLink(VaadinIcon.DASHBOARD, "Dashboard"),
                            createLink(VaadinIcon.CART, "Orders"),
                            createLink(VaadinIcon.USER_HEART, "Customers"),
                            createLink(VaadinIcon.PACKAGE, "Products"));

            return navigation;
    }

    private RouterLink createLink(VaadinIcon icon, String viewName) {
            RouterLink link = new RouterLink();
            // Demo has no routes
            // link.setRoute(viewClass.java);
            link.addClassNames(LumoUtility.Display.FLEX,
                            LumoUtility.AlignItems.CENTER,
                            LumoUtility.Padding.Horizontal.LARGE,
                            LumoUtility.TextColor.SECONDARY);
            link.add(icon.create());
            // hidden-source-line: workaround to make text color work
            link.getElement().setAttribute("href", viewName); // hidden-source-line
            link.getElement().setAttribute("aria-label", viewName);
            return link;
    }

    public static class Exporter extends DemoExporter<AppLayoutBottomNavbar> { // hidden-source-line
    } // hidden-source-line
      // tag::snippet[]
}
// end::snippet[]
